### PR TITLE
Bug 2 - Output format update

### PR DESCRIPTION
**Pull request by zachomedia**
Updated output format:
- Removed &nbsp
- Removed extra spaces
- Removed HTML tages
- Removed headers (Notes: and Reserve:)

Date format updated:

```
[dates] => Array
 (
  [start_time] => 07:00
  [end_time] => 09:50
  [weekdays] => T
  [start_date] => 09/17
  [end_date] => 09/17
 )
```

I have also updated the documentation (Readme) to reflect the latest output changes.

### CODE REVIEW

**Comment by KartikTalwar** [comment]
Hey @zachomedia thanks for sanitizing the output!

This pretty much handles the dates and times nicely except for online courses and TBA classes.

``` php
print_r(parse_schedule('1131', 'PHYS', '380'));
print_r(parse_schedule('1139', 'PHYS', '490'));
```
- For some online courses, the dates show up as **12:00-12:00** with no day. See code comments.
- For [PHYS380](http://www.adm.uwaterloo.ca/cgi-bin/cgiwrap/infocour/salook.pl?sess=1131&subject=PHYS&cournum=380), the classes array has length 2
- We should probably add another boolean field to dates, `is_tba`. Thoughts?

File: Schedule/schedule.php [hunk 1]
```diff
@@ -140,6 +160,65 @@ function parse_schedule($term_id, $faculty, $course, $level = 'under')
...
+function parse_date($strDate) {
+  $date = array('start_time' => '',
+                'end_time'   => '',
+                'weekdays'  => '',
+                'start_date' => '',
+                'end_date'   => '');
+  $match = array();
+  
+  $strDate = beautify($strDate);
+  $dateRegex = "/(\d{2}:\d{2})-(\d{2}:\d{2})(\w+)\s*(?:(\d{2}\/\d{2})-(\d{2}\/\d{2}))?.*/";
```
Comment (KartikTalwar) [inline Schedule/schedule.php]: Having an optional day matcher we can handle the 12:00-12:00 case by by doing `count($match) >= 3`

``` php
$dateRegex = "/(\d{2}:\d{2})-(\d{2}:\d{2})(\w+)?\s*(?:(\d{2}\/\d{2})-(\d{2}\/\d{2}))?.*/";
```

File: Schedule/schedule.php [hunk 2]
```diff
@@ -140,6 +160,65 @@ function parse_schedule($term_id, $faculty, $course, $level = 'under')
...
+  $matchResult = preg_match($dateRegex, $strDate, $match);
+
```
Comment (KartikTalwar) [inline Schedule/schedule.php:236]: TBA check?

``` php
$tbaRegex = preg_match("/TBA/", $strDate, $tbamatches);

$date['tba'] = count($tbamatches) == 1;
```

Comment (KartikTalwar) [inline Schedule/schedule.php:236]: On the same note, we can also handle **Cancelled Section**

http://www.adm.uwaterloo.ca/cgi-bin/cgiwrap/infocour/salook.pl?sess=1139&subject=PHYS&cournum=771&level=grad

**Comment by zachomedia** [comment]
I've made a few updates (from the suggestions above).
- Online courses where times are 12:00-12:00, with no weekday, now appear correct.
- I've corrected the number of classes in the array (this was caused the updates to date output)
- I've added support for TBA and Cancelled classes:

```
[dates] => Array
  (
    [start_time] => 08:30
    [end_time] => 10:20
    [weekdays] => F
    [start_date] =>
    [end_date] =>
    [is_tba] =>
    [is_cancelled] =>
  )
```

**Comment by KartikTalwar** [comment]
Closes #2

**Pull request merged**